packages/components/src/filters/advanced/number-filter.js

```diff
@@ -0,0 +1,184 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, TextControl } from '@wordpress/components';
+import { get, find, partial } from 'lodash';
+import interpolateComponents from 'interpolate-components';
+import classnames from 'classnames';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TextControlWithAffixes from '../../text-control-with-affixes';
+import { formatCurrency } from '../../../../currency/src';
+
+class NumberFilter extends Component {
+	getBetweenString() {
+		return _x(
+			'{{moreThan /}} and {{lessThan /}}',
+			'Numerical range inputs arranged on a single line',
+			'wc-admin'
+		);
+	}
+
+	getLegend( filter, config ) {
+		const inputType = get( config, [ 'input', 'type' ], 'number' );
+		const rule = find( config.rules, { value: filter.rule } ) || {};
+		let filterStr = filter.value || '';
+
+		if ( 'between' === rule.value ) {
+			let [ moreThan, lessThan ] = filterStr.split( ',' );
+
+			if ( 'currency' === inputType ) {
+				moreThan = formatCurrency( moreThan );
+				lessThan = formatCurrency( lessThan );
+			}
+
+			filterStr = interpolateComponents( {
+				mixedString: this.getBetweenString(),
+				components: {
+					moreThan: <Fragment>{ moreThan }</Fragment>,
+					lessThan: <Fragment>{ lessThan }</Fragment>,
```

Comment:
Do these need the `Fragment` wrapper?

Reply:
I believe so - without passing a node, the `mixedString` isn't interpolated. See: https://github.com/Automattic/interpolate-components/blob/master/src/index.es6#L52